compiler/qsc_partial_eval/tests/intrinsics.rs

```diff
@@ -642,3 +642,259 @@ fn call_to_intrinsic_mresetz_adds_callable_and_generates_instruction() {
                 Return"#]],
     );
 }
+
+#[test]
+fn call_to_intrinsic_begin_estimate_caching_with_classical_values_does_not_generate_instructions() {
+    let program = compile_and_partially_evaluate(indoc! {
+        r#"
+        namespace Test {
+            open Microsoft.Quantum.ResourceEstimation;
+            @EntryPoint()
+            operation Main() : Unit {
+                let _ = BeginEstimateCaching("test", 0);
```

Comment:
Since `BeginEstimateCaching` is usually used to affect control flow, the test should verify that this usage works as expected (namely that the function always returns a hard-coded `true` value for branching). Something like this would be good:

```suggestion
                use q = Qubit();
                if BeginEstimateCaching("test", 0) {
                    X(q);
                }
```

Reply:
Updated test.